ambari-server/src/main/java/org/apache/ambari/server/utils/ModuleVersion.java

```diff
@@ -106,5 +107,31 @@ public int compareTo(ModuleVersion other) {
     return result > 0 ? 1 : result < 0 ? -1 : 0;
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    ModuleVersion that = (ModuleVersion) o;
+
+    if (apacheMajor != that.apacheMajor) return false;
+    if (apacheMinor != that.apacheMinor) return false;
+    if (build != that.build) return false;
+    if (hotfix != that.hotfix) return false;
+    if (internalMaint != that.internalMaint) return false;
+    if (internalMinor != that.internalMinor) return false;
 
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = apacheMajor;
```

Comment:
Can be simplified using `Objects.hash()`.

Reply:
@adoroszlai nope, why to box primitives to calculate hash?   it is not effective